src/Models/Services/Utils.ts

```diff
@@ -381,6 +393,187 @@ export function rgbToHex(r, g, b) {
+        : [];
+};
+
+export const getStoredEditorModelMetadata = () => {
+    const oatData = getStoredEditorData();
+    return oatData && oatData.modelsData && oatData.modelsData.modelsMetadata
+        ? oatData.modelsData.modelsMetadata
+        : [];
+};
+
+// Get stored models' namespace OAT-data
+export const getStoredEditorNamespaceData = () => {
+    const oatData = getStoredEditorData();
+    return oatData && oatData.namespace ? oatData.namespace : null;
+};
+
+export const updateModelId = (
+    oldId: string,
+    newId: string,
+    models: DtdlInterface[],
+    modelPositions: IOATModelPosition[]
+) => {
+    // Update the modelPositions
+    const modelsPositionsCopy = deepCopy(modelPositions);
+
+    // Find the model position with the same id
+    const modelPosition = modelsPositionsCopy.find((x) => x['@id'] === oldId);
+    if (modelPosition) {
+        modelPosition['@id'] = newId;
+    }
+
+    // Update models
+    const modelsCopy = deepCopy(models);
+    const modelCopy = modelsCopy.find((x) => x['@id'] === oldId);
+    if (modelCopy) {
+        modelCopy['@id'] = newId;
+    }
+
+    // Update contents
+    modelsCopy.forEach((m) =>
+        m.contents.forEach((c) => {
+            const r = c as DtdlRelationship;
+            if (r && r.target === oldId) {
+                r.target = newId;
+            }
+            if (r && r['@id'] === oldId) {
+                r['@id'] = newId;
+            }
+
+            const p = c as DtdlInterfaceContent;
+            if (p && p.schema === oldId) {
+                p.schema = newId;
+            }
+
+            if (m.extends) {
+                const e = m.extends as string[];
+                const i = e.indexOf(oldId);
+                if (i >= 0) {
+                    e[i] = newId;
+                }
+            }
+        })
+    );
+
+    return [modelsCopy, modelsPositionsCopy];
+};
+
+// Get fileName from DTMI
+export const getFileNameFromDTMI = (dtmi: string) => {
+    // Get id path - Get section between last ":" and ";"
+    const initialPosition = dtmi.lastIndexOf(':') + 1;
+    const finalPosition = dtmi.lastIndexOf(';');
+
+    if (initialPosition !== 0 && finalPosition !== -1) {
+        const idPath = dtmi.substring(initialPosition, finalPosition);
+        const idVersion = dtmi.substring(
+            dtmi.lastIndexOf(';') + 1,
+            dtmi.length
+        );
+        return `${idPath}-${idVersion}`;
+    }
+};
+
+// Get directoryPath from DTMI
+export const getDirectoryPathFromDTMI = (dtmi: string) => {
+    const initialPosition = dtmi.indexOf(':') + 1;
+    const finalPosition = dtmi.lastIndexOf(':');
+
+    if (initialPosition !== 0 && finalPosition !== -1) {
+        const directoryPath = dtmi.substring(initialPosition, finalPosition);
+        // Scheme - replace ":" with "\"
+        return directoryPath.replace(':', '\\');
+    }
+};
+
+// Load files from local storage
+export const loadFiles = () =>
```

Comment:
same here, do we have types defined for this? Mabye just add it to the function definition if we do.